components/global/mixer-adapter/pkg/wso2spadapter/wso2spadapter.go

```diff
@@ -60,53 +52,94 @@ type (
 
 	// Wso2SpAdapter supports metric template.
 	Wso2SpAdapter struct {
-		listener net.Listener
-		server   *grpc.Server
-		logger   *zap.SugaredLogger
+		listener         net.Listener
+		server           *grpc.Server
+		logger           *zap.SugaredLogger
+		httpClient       *http.Client
+		metricsPublisher MetricsPublisher
+		spServerUrl      string
+	}
+
+	/* This interface and the struct has been implemented to test the publishMetrics() function */
+	MetricsPublisher interface {
+		publishMetrics(attributeMap map[string]interface{}, logger *zap.SugaredLogger, httpClient *http.Client, spServerUrl string) bool
```

Comment:
Either make publishMetrics public or make MetricsPublisher private. Otherwise, we are exposing an empty interface

Reply:
You can create interface similar to this
Publisher interface {
   Publish(...) bool
}